config/routes.rb

```diff
@@ -2,60 +2,81 @@
-  post '/vms/vm/:id/reboot_guest_os' => 'vms#reboot_guest_os', constraints: { id: /.*/ }
-  post '/vms/vm/:id/reset_vm' => 'vms#reset_vm', constraints: { id: /.*/ }
-  post '/vms/vm/:id/request_vm_archivation' => 'vms#request_vm_archivation', constraints: { id: /.*/ }
-  post '/vms/vm/:id/archive_vm' => 'vms#archive_vm', constraints: { id: /.*/ }
-  post '/vms/vm/:id/request_vm_revive' => 'vms#request_vm_revive', constraints: { id: /.*/ }
-  post '/vms/vm/:id/revive_vm' => 'vms#revive_vm', constraints: { id: /.*/ }
-  post '/vms/vm/:id/stop_archiving' => 'vms#stop_archiving', constraints: { id: /.*/ }
-  resources :vms, except: %i[new create], path: 'vms/vm'
-
-  get 'slack/new' => 'slack#new', as: :new_slack
-  get 'slack/auth' => 'slack#update', as: :update_slack
-
-  devise_for :users,
-             path: 'users',
-             controllers: {
-               registrations: 'users/registrations',
-               omniauth_callbacks: 'users/omniauth_callbacks'
-             }
+  # '/slack/...'
+  scope 'slack' do
+    get 'new' => 'slack#new', as: :new_slack
+    get 'auth' => 'slack#update', as: :update_slack
+  end
 
-  resources :hosts, :servers
-  resources :users do
-    member do
-      patch :update_role
+  # https://guides.rubyonrails.org/routing.html#prefixing-the-named-route-helpers
+  # '/vms/...'
+  scope 'vms' do
+    # '/vms/request_templates/...'
+    resources :request_templates, except: :show
+    # '/vms/requests/operating_systems/...'
+    # Order matters here, as routes are matched from top to bottom
+    # If 'resources :requests' is first, it will attempt to match that
+    resources :operating_systems, path: 'requests/operating_systems', except: :show
+    # '/vms/requests/...'
+    resources :requests do
+      post :push_to_git, on: :member
+      patch :reject, on: :collection
+    end
+    # '/vms/configs/:id'
+    scope 'configs' do
+      get ':id' => 'vms#edit_config', constraints: { id: /.*/ }, as: :edit_config
+      patch ':id' => 'vms#update_config', constraints: { id: /.*/ }, as: :update_config
+    end
+    # '/vms/vm'
+    # move all vm actions under /vms/vm because a VM named requests might otherwise lead to issues!
+    resources :vms, except: %i[new create], path: 'vm', constraints: { id: /.*/ } do
+      # https://guides.rubyonrails.org/routing.html#adding-member-routes
+      member do
+        post 'change_power_state'
+        post 'suspend_vm'
+        post 'shutdown_guest_os'
+        post 'reboot_guest_os'
+        post 'reset_vm'
+        post 'request_vm_archivation'
+        post 'archive_vm'
+        post 'request_vm_revive'
+        post 'revive_vm'
+        post 'stop_archiving'
+      end
     end
   end
 
-  resources :projects, only: %i[index show new edit create update]
-
-  root 'landing#index'
+  # '/users/...'
+  # https://github.com/plataformatec/devise#configuring-routes
+  devise_for :users, path: 'users',
```

Comment:
Why do we need path: 'users'?

Reply:
I believe 'users' is the default for Devise anyway, not too bad to have it in the `routes.rb` as a reminder for the employed path and where to change it.

https://www.rubydoc.info/github/plataformatec/devise/ActionDispatch%2FRouting%2FMapper:devise_for

Reply:
That was the reason for my comment, but the argumentation to define the default value as a reminder I do not understand.